pkg/state/state_test.go

```diff
@@ -4088,3 +4088,104 @@ func TestHelmState_chartOCIFlags(t *testing.T) {
 		})
 	}
 }
+
+func TestIsOCIChart(t *testing.T) {
+	cases := []struct {
+		st       *HelmState
+		chart    string
+		expected bool
+	}{
+		{&HelmState{}, "oci://myrepo/mychart", true},
+		{&HelmState{}, "oci://myrepo/mychart:1.0.0", true},
+		{&HelmState{}, "myrepo/mychart", false},
+		{&HelmState{}, "myrepo/mychart:1.0.0", false},
+		{
+			&HelmState{
+				ReleaseSetSpec: ReleaseSetSpec{
+					Repositories: []RepositorySpec{
+						{
+							Name: "ocirepo",
+							URL:  "ocirepo.com",
+							OCI:  true,
+						},
+					},
+				},
+			},
+			"ocirepo/chart",
+			true,
+		},
```

Comment:
```suggestion
		},
		{
			&HelmState{
				ReleaseSetSpec: ReleaseSetSpec{
					Repositories: []RepositorySpec{
						{
							Name: "nonocirepo",
							URL:  "nonocirepo.com",
						},
					},
				},
			},
			"nonocirepo/chart",
			false,
		},
```

Reply:
Sorry for back and forth! But to be extra clear, I think this case fail until https://github.com/helmfile/helmfile/pull/1661/files#r1741363322 is addressed

Reply:
@mumoshu yeah. got it.